yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,162 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    float_or_none,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class AcFunVideoIE(InfoExtractor):
+    IE_NAME = 'AcFunVideoIE'
+    _VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/v/)ac(?P<id>[_\d]+)'
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'info_dict': {
+            'id': '35283657',
+            'title': '【十五周年庆】文章区UP主祝AcFun生日快乐！',
+            'duration': 455.21,
+            'timestamp': 1655289827,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'only_matching': True,
+    }]
+
+    @staticmethod
+    def parse_format(info):
+        vcodec, acodec = None, None
+
+        codec_str = info.get('codecs') or ''
+        m = re.match('(?P<vc>[^,]+),(?P<ac>[^,]+)', codec_str)
+        if m:
+            vcodec = m.group("vc")
+            acodec = m.group("ac")
+
+        return {
+            'url': info.get('url'),
+            'fps': int_or_none(info.get('frameRate')),
+            'width': int_or_none(info.get('width')),
+            'height': int_or_none(info.get('height')),
+            'vcodec': vcodec,
+            'acodec': acodec,
+            'tbr': float_or_none(info.get('avgBitrate'))
+        }
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
```

Comment:
i think the `mobj` is unnecessary as the url will be get checked first, you can group `video_id` directly here

Reply:
en, you mean to call self._match_id(url) ?

Reply:
> en, you mean to call self._match_id(url) ? the _VALID_URL almost have nothing else. `_VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/v/)ac(?P<id>[_\d]+)'`

Yes

Reply:
ok